lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3554,7 +3555,8 @@ static Type resolveDependentMemberTypes(GenericSignatureBuilder &builder,
         equivClass->recursiveConcreteType = false;
```

Comment:
Somewhat unrelated: Should we be setting this in maybeResolveEquivalenceClass() as well?

Reply:
I was actually wrong, there is no infinite recursion on master. This is cherry-picked from a branch that relaxes recursive superclass constraints, which happens to be the cause of the recursion this is supposed to fix, but I misunderstood how `not` works and running into the same problem with the 5.1.3 release convinced me the infinite recursion is present on master as well, although I didn't even bother to check.

So, answering your question – probably yes, but it seems we will not have something to test against until after we lift the restriction on these recursive constraints.

Still doesn't hurt having this though.

Reply:
I would prefer it if the crashers in validation-tests did not check for precise diagnostic messages, and instead asserted that the compiler rejected the code without crashing (which is what `not swift -frontend -typecheck` accomplishes)